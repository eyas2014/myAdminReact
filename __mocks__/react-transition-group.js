import React from 'react';

let result=[]
function isCalledWith(){

	return result;
};

function reset(){

	result=[];
};

function CSSTransition(props){
	const {classNames, timeout, unmountOnExit}=props;
	result.push({in: props.in, classNames, timeout, unmountOnExit});
	return (props.children);
}



export {CSSTransition, isCalledWith, reset};