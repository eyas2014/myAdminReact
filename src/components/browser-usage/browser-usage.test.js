import React from 'react';
import BrowserUsage from './browser-usage.jsx';
import enzyme from 'enzyme';

jest.mock('react-transition-group');
jest.mock('react-chartjs-2');

 var PieData = {
		datasets: [{ data	    : [700, 500, 400, 600, 300, 100],
			     backgroundColor: ['#f56954', '#00a65a',  '#f39c12', '#00c0ef', '#3c8dbc', '#d2d6de'] }],
		};

var options={cutoutPercentage: 45 };

document.body.innerHTML='<div  id="root"></div>';

const mountWrapper=enzyme.mount(<BrowserUsage data={{data:PieData, options}}></BrowserUsage>);

it('should render correct htmls',()=>{
	expect(mountWrapper.html()).toMatchSnapshot();
});

it('chartjs should be called with correct arguments', ()=>{
	expect(require('react-chartjs-2').isCalledWith()).toEqual({data: PieData, options})
});

it('should render correct number of legends', ()=>{
	expect(mountWrapper.find('.browser-usage main ul li').length).toEqual(6);	
});

it('should render correct number of stat panels', ()=>{
	// expect(mountWrapper.find('.browser-usage footer>div').length).toEqual(3);	
});

it('should collapse after clicking toggle', ()=>{
	require('react-transition-group').reset();
	mountWrapper.find('.browser-usage header').childAt(1).childAt(0).simulate('click');
	expect(require('react-transition-group').isCalledWith()[0].classNames).toEqual('componentShutdown');
	expect(require('react-transition-group').isCalledWith()[0].in).toEqual(true);
	expect(require('react-transition-group').isCalledWith()[1].classNames).toEqual('componentCollapse');
	expect(require('react-transition-group').isCalledWith()[1].in).toEqual(false);
});

it('should expand after clicking toggle', ()=>{
	require('react-transition-group').reset();
	mountWrapper.find('.browser-usage header').childAt(1).childAt(0).simulate('click');
	expect(require('react-transition-group').isCalledWith()[0].classNames).toEqual('componentShutdown');
	expect(require('react-transition-group').isCalledWith()[0].in).toEqual(true);
	expect(require('react-transition-group').isCalledWith()[1].classNames).toEqual('componentCollapse');
	expect(require('react-transition-group').isCalledWith()[1].in).toEqual(true);
});

it('should close after clicking close', ()=>{
	require('react-transition-group').reset();
	mountWrapper.find('.browser-usage header').childAt(1).childAt(1).simulate('click');
	expect(require('react-transition-group').isCalledWith()[0].classNames).toEqual('componentShutdown');
	expect(require('react-transition-group').isCalledWith()[0].in).toEqual(false);
});
