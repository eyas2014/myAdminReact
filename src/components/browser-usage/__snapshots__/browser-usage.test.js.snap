// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct htmls 1`] = `"<div class=\\"browser-usage\\"><header class=\\"box-header box-border-bottom clearfix\\"><div class=\\"btn\\"><span>Browser Usage</span></div><div class=\\"btn-group float-right\\"><div class=\\"btn\\"><i class=\\"fas fa-minus text-muted\\"></i></div><div class=\\"btn\\"><i class=\\"fas fa-times text-muted\\"></i></div></div></header><div class=\\"wrapper\\"><main class=\\"box-border-bottom\\"><div class=\\"chart-responsive\\"><div id=\\"doughnut-chart\\"></div></div><div><ul class=\\"p-0\\"><li class=\\"ml-2 d-inline-block\\"><i class=\\"text-danger far fa-circle\\"></i><span class=\\"ml-1\\">Chrome</span></li><li class=\\"ml-2 d-inline-block\\"><i class=\\"text-success far fa-circle\\"></i><span class=\\"ml-1\\">IE</span></li><li class=\\"ml-2 d-inline-block\\"><i class=\\"text-warning far fa-circle\\"></i><span class=\\"ml-1\\">FireFox</span></li><li class=\\"ml-2 d-inline-block\\"><i class=\\"text-primary far fa-circle\\"></i><span class=\\"ml-1\\">Safari</span></li><li class=\\"ml-2 d-inline-block\\"><i class=\\"text-info far fa-circle\\"></i><span class=\\"ml-1\\">Opera</span></li><li class=\\"ml-2 d-inline-block\\"><i class=\\"text-muted far fa-circle\\"></i><span class=\\"ml-1\\">Navigator</span></li></ul></div></main><footer class=\\"\\"><div class=\\"box-border-bottom\\"><div class=\\"btn\\"><span>United States of America</span></div><div class=\\"float-right btn text-danger\\"><i class=\\"fa fa-angle-down font-weight-light\\"></i><span class=\\"\\">12%</span></div></div><div class=\\"box-border-bottom\\"><div class=\\"btn\\"><span>India</span></div><div class=\\"float-right btn text-success\\"><i class=\\"fa fa-angle-up font-weight-light\\"></i><span class=\\"\\">12%</span></div></div><div class=\\"box-border-bottom\\"><div class=\\"btn\\"><span>China</span></div><div class=\\"float-right btn text-warning\\"><i class=\\"fa fa-angle-left font-weight-light\\"></i><span class=\\"\\">12%</span></div></div></footer></div></div>"`;
